lib/install/phases/coredns.go

```diff
@@ -163,8 +145,26 @@ func (r *corednsExecutor) Rollback(context.Context) error {
 
 // GenerateCorefile will generate a coredns configuration file to be used from within the cluster
 func GenerateCorefile(config CorednsConfig) (string, error) {
+	// Read the resolv.conf from the host doing installation // upgrade
+	// it will be used for configuring coredns upstream servers
+	resolvConf, err := systeminfo.ResolvFromFile("/etc/resolv.conf")
+	if err != nil {
+		return "", trace.Wrap(err)
+	}
+
+	// Optionally try and load upstream nameservers from systemd-resolved by reading the compatibility resolv.conf
+	// More Info: https://github.com/gravitational/gravity/issues/606#issuecomment-529171440
+	// TODO(knisbet) is there a better way to pull upstream resolvers directly from systemd?
+	systemdResolvConf, err := systeminfo.ResolvFromFile("/run/systemd/resolve/resolv.conf")
+	if err != nil && !trace.IsNotFound(err) {
+		return "", trace.Wrap(err)
+	}
+
+	config.UpstreamNameservers = mergeUpstreamResolvers(resolvConf, systemdResolvConf)
+	config.Rotate = resolvConf.Rotate || systemdResolvConf.Rotate
```

Comment:
This is going to use `systemdResolvConf` even when it's not been initialized (i.e. the file was not read)

Reply:
I'm not sure I understand this concern. An uninitialized config will have the rotate bool as false. A boolean or of false will stay false. So whichever config we're able to pick up and merge, if it's set on one it will be set on both. I suppose there is an issue if both /etc/resolv and the systemd resolv both specify upstreams that are configured in a different way, but for now we're just merging the configs, we don't have a good way to dictate which we should be using.

Reply:
My concern is the potential nil pointer access.

Reply:
IC, yea I missed that ResolvFromFile returns nil and not an empty struct.